compiler/rustc_hir_analysis/src/check/region.rs

```diff
@@ -885,3 +892,190 @@ pub fn region_scope_tree(tcx: TyCtxt<'_>, def_id: DefId) -> &ScopeTree {
+            && let Some(body_id) = tcx.hir().maybe_body_owned_by(item.owner_id.def_id)
+            && let body = tcx.hir().body(body_id)
+            && let hir::ExprKind::Block(blk, _) = body.value.kind
+        {
+            check_stmts(tcx, body_id.hir_id, blk.stmts, safe);
+            return;
+        } else {
+            return;
+        };
+
+        match (init.kind, loc.pat.kind) {
+            (hir::ExprKind::AddrOf(borrow_kind, m, expr), _) => {
+                if matches!(borrow_kind, BorrowKind::Ref)
+                    && let hir::ExprKind::Path(qpath) = expr.kind
+                    && let hir::QPath::Resolved(_, path) = qpath
+                    && let hir::def::Res::Def(def_kind, _) = path.res
+                    && let hir::def::DefKind::Static(mt) = def_kind
+                    && matches!(mt, Mutability::Mut)
+                {
+                    handle_static_mut_ref(
+                        tcx,
+                        init.span,
+                        qpath_to_string(&qpath),
+                        loc.span.edition().at_least_rust_2024(),
+                        matches!(m, Mutability::Mut),
+                        safe,
+                        hir_id,
+                    );
+                }
+            }
+            (hir::ExprKind::Path(qpath), hir::PatKind::Binding(ba, _, _, _)) => {
+                if matches!(ba.0, rustc_ast::ByRef::Yes)
+                    && let hir::QPath::Resolved(_, path) = qpath
+                    && let hir::def::Res::Def(def_kind, _) = path.res
+                    && let hir::def::DefKind::Static(mt) = def_kind
+                    && matches!(mt, Mutability::Mut)
+                {
+                    handle_static_mut_ref(
+                        tcx,
+                        init.span,
+                        qpath_to_string(&qpath),
+                        loc.span.edition().at_least_rust_2024(),
+                        matches!(ba.1, Mutability::Mut),
+                        safe,
+                        hir_id,
+                    );
+                }
+            }
+            (hir::ExprKind::Tup(exprs), _) => {
+                for expr in exprs.iter() {
+                    maybe_expr_static_mut(
+                        tcx,
+                        *expr,
+                        expr.span,
+                        expr.span.edition().at_least_rust_2024(),
+                        safe,
+                        hir_id,
+                    );
+                }
+            }
+            _ => {}
+        }
+    }
+
+    fn handle_static_mut_ref(
+        tcx: TyCtxt<'_>,
+        span: Span,
+        var: String,
+        e2024: bool,
+        mutable: bool,
+        safe: bool,
+        hir_id: HirId,
+    ) {
+        if e2024 {
+            let mut sugg = errors::StaticMutRefSugg::Shared { span, var: var.clone() };
+            if mutable {
+                sugg = errors::StaticMutRefSugg::Mut { span, var };
+            }
```

Comment:
```suggestion
            let sugg = if mutable {
                errors::StaticMutRefSugg::Mut { span, var }
            } else {
                errors::StaticMutRefSugg::Shared { span, var: var.clone() }
            };
```
nit: this is just a little tidier